File: src/ClientModel/Constructor.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

using System.Collections.Generic;

namespace AutoRest.Java.Model
{
    /// <summary>
    /// The constructor in a ServiceClient.
    /// </summary>
    public class Constructor
    {
        public Constructor(params ClientMethodParameter[] parameters)
        {
            Parameters = parameters;
        }

        /// <summary>
        /// The parameters of this constructor.
        /// </summary>
        public IEnumerable<ClientMethodParameter> Parameters { get; }

        public void AddImportsTo(ISet<string> imports, bool includeImplementationImports)
        {
            foreach (ClientMethodParameter parameter in Parameters)
            {
                parameter.AddImportsTo(imports, includeImplementationImports);
            }
        }
    }
}

File: mapper/src/main/java/com/azure/autorest/model/clientmodel/Constructor.java
package com.azure.autorest.model.clientmodel;

import java.util.List;
import java.util.Set;

// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.


/** 
 The constructor in a ServiceClient.
*/
public class Constructor
{
    public Constructor(List<ClientMethodParameter> parameters)
    {
        Parameters = parameters;
    }

    /**
     The parameters of this constructor.
    */
    private List<ClientMethodParameter> Parameters;
    public final List<ClientMethodParameter> getParameters()
    {
        return Parameters;
    }

    public final void AddImportsTo(Set<String> imports, boolean includeImplementationImports)
    {
        for (ClientMethodParameter parameter : getParameters())
        {
            parameter.AddImportsTo(imports, includeImplementationImports);
        }
    }
}